File: Java/Arrays/BubbleSort/BubbleSort.java

import java.util.Arrays;

public class BubbleSort {

	public static void main(String[] args) {

		int[] arr = {1,4,2,4,5,3,5,7,3};

		for (int i = arr.length - 1; i > 0; i--) {
			for (int j = 0; j < i; j++) {
				if (arr[j] > arr[j + 1]) {
					swap(arr, j, j + 1);
				}
			}
		}

		System.out.println(Arrays.toString(arr));

	}

	public static void swap(int[] arr, int i, int j) {
		int a = arr[i];
		arr[i] = arr[j];
		arr[j] = a;
	}
}

File: JavaScript/Arrays/BubbleSort/bubblesort.js
var arr = [3,445,3,654,6,4568,5479,46,3456,2357,368,45,367,248,357,935,687346,835];

function bubbleSort(a) {
	for (var i = arr.length - 1; i > 0; i--) {
		for (var j = 0; j < i; j++) {
			if (arr[j] > arr[j + 1]) {
				swap(arr, j, j + 1);
			}
		}
	}
}

function swap(arr, i, j) {
	var a = arr[i];
	arr[i] = arr[j];
	arr[j] = a;
}


bubbleSort(arr);
alert(arr)